src/test-kit/OptionButtons/sortByButton-driver.ts

```diff
@@ -0,0 +1,40 @@
+import { Locator, Page } from 'playwright-chromium';
+import { selectors } from '../../test-kit/OptionButtons/selectors';
+
+export class SortByButtonDriver {
+   private sortByButton: Locator;
+   private dropDownOptionList: Locator;
+
+   constructor(private page: Page) {
+      this.sortByButton = page.locator(selectors.sortByButton);
+      this.dropDownOptionList = page.locator(
+         selectors.sortbyButtonDropDownList
+      );
+   }
+
+   async toggleDropdown() {
+      await this.sortByButton.click();
+   }
```

Comment:
Too specific function, it will make the test longer:
```ts
await toggleDropdown();
await clickOption(...);
```

instead just do this:
```ts
await changeSortToBe(...);
```

That way it is clearer what we are doing and also shorter, notice that in the first option (that you did), you can't actually know what the dropdown changing from the test itself.

Reply:
I understand that adding the function `changeSortToBe(...)` could be beneficial. However, it’s important to note that I use `await toggleDropdown();` without `await clickOption(...);` to first check if the dropdown list is visible.
For example:
```markdown
   it('should toggle dropdown options list on button click', async function () {
      expect(await sortByButtonDriver.isDropdownVisible()).to.be.false;
      await sortByButtonDriver.toggleDropdown();
      expect(await sortByButtonDriver.isDropdownVisible()).to.be.true;
      await sortByButtonDriver.toggleDropdown();
      expect(await sortByButtonDriver.isDropdownVisible()).to.be.false;
   });
   ```